File: csharp-projects/Leetcode/0011-container-with-most-water/0011-container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        area = 0
        leftPointer = 0
        rightPointer = len(height) - 1
        while(leftPointer < rightPointer):
            h = height[leftPointer] if height[leftPointer] < height[rightPointer] else height[rightPointer]
            area = h * (rightPointer - leftPointer) if area < h * (rightPointer - leftPointer) else area
            if height[leftPointer] <= height[rightPointer]:
                leftPointer += 1
            else:
                rightPointer -= 1
        return area

File: csharp-projects/Leetcode/0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let area = 0
    let leftPointer = 0;
    let rightPointer = height.length - 1;
    while(leftPointer < rightPointer) {
        let tempHeight = Math.min(height[leftPointer],height[rightPointer]);
        let tempArea = tempHeight*(rightPointer-leftPointer);
        area = Math.max(area,tempArea)
        if (height[leftPointer] <= height[rightPointer]) leftPointer++;
        else rightPointer--;
    }
    return area;
};